Fix format for dates on copy

jdrossl opened the pull request:
https://github.com/craftercms/craftercms/issues/5268

sumerjabri reviewed (changes requested):
Can this be made a common utility function that's used wherever this is needed to avoid this type of error in the future?

jdrossl commented:
This is the only place so far where this format is required, but yeah in the future it is possible to reuse it

sumerjabri commented:
Please centralize it in the utils/helper class as a static method, add Javadoc, and use the shared helper.

jdrossl committed:
Fix for date format:
- Centralize date handling in a utility class
- Updated all existing uses with the new utility
